README.md

```diff
@@ -22,11 +22,12 @@ df = DataFrame(species=["E. coli", "B. fragilis", "L. casei"],
                sample2=[0.3, 0.7, 0.0],
                sample3=[0.0, 0.2, 0.8])
 
-abund = AbundanceTable(df)
+abund = abundancetable(df)
 
 @show abund
-@show abund.samples
-@show abund.features
+full(abund)
+@show sitenames(abund)
+@show specnames(features)
```

Comment:
Three things to talk about here: 
1. Of course the names of the internal fields are different. AbundanceTable has feature and samples, ComMatrix has species and sites. I think I like the generality of samples, though it may be confused with the `sample` function, but `feature` is not so intuitive for me.
2. One way of adressing this is to use a getter function rather than the field name - e.g. if I kept the names in ComMatrix, we could just define `samples(x) = sitenames(x)` and then have `samples(abund)`, getting specificity without losing generality.
3. I'm not sure why you use `@show`?

Reply:
1. "Feature" is kind of a machine learning thing, I think. Lots of folks in the lab talk about it in these terms, which is why I chose it. I'd like something more generic than specnames because some tables have species, some have gene functions, some have metabolic pathways etc. All have ≈ the same structure and a lot of the same functions (reads mapped to a particular pathway, as counts or relative abundance etc). Likewise, "sitenames" doesn't make much sense either in this context, unless the site you're talking about is a person's colon :-P
2. I think I'd be fine with this implementation
3. No reason in particular... in general I just like it better than `println()`? Happy to entertain alternatives

Reply:
I think it could make sense, even at the macro scale, to talk about `sample`, though `community` might be even better? What about `taxon` instead of species, could work for OTUs etc. I'm not sure gene functions or metabolic pathways should come in a different format, like in a trait matrix?

Reply:
I'm drawn by community, though I think it might be less intelligible to someone new coming to this. Samples is the most intuitive for my use-cases, though I agree that it might be wise to avoid confusion with the statistics function of the same name.

Taxon also doesn't work for a number of my use-cases, but those might be idiosyncratic and if we're going to end up doing a thin wrapper around AxisArrays as you're talking about over at richardreeve/Diversity.jl/issues/23, then maybe I can just call them something different depending on the context and just use aliases? Ultimately, all of my operations are just on matrices where it's nice to store the name of the columns and the rows.

Reply:
I'll start implementing the change based on AxisArrays one of these days, and also transpose the internal matrix, then update this PR to reflect.